pkg/common/providers/ocmprovider/cluster.go

```diff
@@ -360,16 +361,17 @@ func (o *OCMProvider) DetermineMachineType(cloudProvider string) (string, error)
 
 		for range machinetypes.Items().Slice() {
 			machinetypeObj := machinetypes.Items().Slice()[rand.Intn(machinetypes.Total())]
-			computeMachineType = machinetypeObj.ID()
+			returnedType = machinetypeObj.ID()
 			break
 		}
+	}
 
-		log.Printf("Random machine type requested, selected %s machine type.", computeMachineType)
+	log.Printf("Random machine type requested, selected `%s` machine type.", returnedType)
 
-		// Update the Config with the selected random region
-		viper.Set(ComputeMachineType, computeMachineType)
-	}
-	return computeMachineType, nil
+	// Update the Config with the selected random region
+	viper.Set(ComputeMachineType, returnedType)
+
```

Comment:
One or both of these statements seem like they may belong inside of the `if` block.